README.md

```diff
@@ -1,2 +1,113 @@
+| excludes | -e, --excludes | Array of Strings | Translation directory regex pattern(s) to glob exclude from the search patterns. | [ ] |
+| outputFileSystem | N/A | File System Module | The filesystem to use to write the translation and loader files. Note: The file system provide must support `mkdirp`. | [fs-extra](https://www.npmjs.com/package/fs-extra) |
+| locales  | -l, --locales | Array of Strings | The list of locale codes to aggregate. **Note: 'en' is always added if not specified.** | [terra-supported locales](https://github.com/cerner/terra-core/blob/master/packages/terra-i18n/src/i18nSupportedLocales.js) |
+| outputDir | -o, --ouputDir | String | Output directory for the translation and loader files | ./aggregated-translations |
+| configPath | -c, --config | String | The path to the terra i18n configuration file | undefined |
+| format | -f, --format | String | The format of syntax to output the translations with. Possible values are 'es5' and 'es6' | 'es5' |
+
+#### Setup Example
+The `aggregate-translations` setup function can be used as follows:
+```js
+// webpack config file
+const aggregateTranslations = require('terra-aggregate-translations');
+
+const aggregateOptions = {
+    baseDir: __dirname,
+    excludes: ['./node_modules/packageToExclude'],
+    locales: ['en', 'en-US'],
+    format: 'es6',
+};
+
+aggregateTranslations(aggregateOptions);
+
+module.exports = // ...webpack config;
+```
+
+#### CLI Example
+The `aggregate-translations` CLI is supplied as a bin script, called `tt-aggregate-translations`, and can be used as follows:
+```js
+scripts: {
+    // ...other scripts
+    "aggregate-translations": "tt-aggregate-translations -l ['en','es'] -e ./node_modules/packageToExclude",
+    "start:build": "npm run aggregate-translations && npm run start"
+}
+```
+
+#### terraI18n.config Example
+Both the setup function and CLI methods allow for the configuration options to be loaded via a config file. The `aggregate-translations` tool will always attempt to load this config by default. If no config path is provided, the aggregate-translations tool will attempt to load the configuration from `./terraI18n.config.js`. If this file does not exist, the default configuration is used.
+
+Add a terra-i18n config file like:
+```js
+module.exports = {
+  locales: ['en', 'en-US'],
+};
+```
+
+Then, to load the config someplace other than `./terraI18n.config.js`, simply add the config path as follows:
+```js
+// using setup approach
+aggregateTranslations({ configPath: './config/terraI18n.config.js' });
+
+// using CLI approach
+"aggregate-translations": "tt-aggregate-translations -c ./config/terraI18n.config.js",
+```
+
+## Resolving Translations and Loaders
+To provide the aggregated-translations files and loaders as modules to the terra-i18n component, the translated output directory must be supplied the webpack `resolve.modules` key before 'node_modules' resolution:
```

Comment:
What does this line mean? 

> 'the translated output directory must be supplied the webpack `resolve.modules` key before 'node_modules' resolution'

Reply:
@nramamurth 

Since webpack is a static code analysis tool used to bundle assets, webpack need to know where imported modules are coming from. So it uses/searches through the resolve.modules directories defined in the webpack configuration when bundling assets. Because we import the translation loader, loacle loader and locale file for a specific locale, we need to include the aggregate-translations directory in the `resolve.modules` list. Then since the node_modules dir is huge and because we want to honor our project's aggregated-translations (incase it was exported in another project) we need to list it first.

more info: https://webpack.js.org/concepts/module-resolution/